extensions/antora/xref-hash-validator/xref-hash-validator.js

```diff
@@ -0,0 +1,142 @@
+
+                if (href && !href.startsWith('http') && !href.startsWith('/docs')) {
+                    return [href]
+                }
+                return []
+            })
+
+        })
+        
+        // we can go back into the contentCatalog and check the xref data we just added
+        // for every file in the contentCatalog, check if the xrefChecker.internalLinks are valid
+        // by verifying that for an internal link to a section, the hash does actually exist in the target page
+        files.forEach( (file) => {
+            if (!file.out || !file.asciidoc) return
+    
+            // what is the full url of each target?
+            file.xrefChecker.internalLinks.forEach((link) => {
+
+                searchForThisURL = new URL(path.join(file.pub.url, link), playbook.site.url)
+
+                // if there's no hash the xref is already checked by Antora
+                // this check shouldn't ever match anyway because we should have already filtered out links without hashes
+                if (!searchForThisURL.hash) return
+                
+                // Find the file in the contentcatalog that has a file.pub.url value that matches the pathname of the link target
+                // where the pathname is a relative path from the site root, which is also what pub.url represents
+                const targetFile = contentCatalog
+                .findBy({ family: 'page' })
+                .filter((page) => page.pub)
+                .filter((f) => {
+                    return f.pub.url === searchForThisURL.pathname
+                })[0]
+
+                const hashTarget = decodeURI(searchForThisURL.hash.replace('#', ''))
+
+                // if the anchor doesn't exist anywhere...
+                if (!linkTargets[hashTarget]) {
+
+                    if (linkTargets[hashTarget.replace(' ', '-')]) {
+                        logger[logLevel]({ file: file.src, source: file.src.origin }, 'anchor %s not found in target page %s - did you mean %s?', decodeURI(searchForThisURL.hash), targetFile.src.path, hashTarget.replace(' ', '-'))
+                        return
+                    }
+                    
+                    // default generated id exists?
+                    if (linkTargets[`_${hashTarget.replace('-', '_')}`]) {
+                        logger[logLevel]({ file: file.src, source: file.src.origin }, 'anchor %s not found in target page %s - did you mean the default generated ID %s?', decodeURI(searchForThisURL.hash), targetFile.src.path, `_${hashTarget.replace('-', '_')}`)
+                        return
+                    }
+                    
+                    // lower case?
+                    if (linkTargets[hashTarget.toLowerCase()]) {
+                        logger[logLevel]({ file: file.src, source: file.src.origin }, 'anchor %s not found in target page %s - use %s', decodeURI(searchForThisURL.hash), targetFile.src.path, searchForThisURL.hash.toLowerCase())
```

Comment:
```suggestion
                        logger[logLevel]({ file: file.src, source: file.src.origin }, 'anchor %s not found in target page %s - did you mean it lowercase %s ?', decodeURI(searchForThisURL.hash), targetFile.src.path, searchForThisURL.hash.toLowerCase())
```
I like your way of specifying how the suggestion differs from what was provided.